zuul/src/main/java/org/egov/filters/route/ChatbotRouter.java

```diff
@@ -0,0 +1,126 @@
+package org.egov.filters.route;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.netflix.zuul.ZuulFilter;
+import com.netflix.zuul.context.RequestContext;
+import com.netflix.zuul.exception.ZuulException;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.net.URL;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+@Component
+@ConditionalOnProperty( value = "home.isolation.chatbot.router.enabled", havingValue = "true")
+public class ChatbotRouter extends ZuulFilter {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${chatbot.context.path}")
+    private String chatbotContextPath;
+
+    @Value("${egov.user.isolation.service.host}")
+    private String isolationUserServiceHost;
+    @Value("${egov.user.isolation.service.search.path}")
+    private String userSearchPath;
+
+    @Value("${egov.statelevel.tenant}")
+    public String stateLevelTenantId;
+
+    @Value("${home.isolation.chatbot.host}")
+    private String homeIsolationChatbotHost;
+
+    @Override
+    public String filterType() {
+        return "route";
+    }
+
+    @Override
+    public int filterOrder() {
+        return 0;
+    }
+
+    @Override
+    public boolean shouldFilter() {
+        String uri = RequestContext.getCurrentContext().getRequest().getRequestURI();
+        return uri.contains(chatbotContextPath);
+    }
+
+    @SneakyThrows
+    @Override
+    public Object run() throws ZuulException {
+        RequestContext context = RequestContext.getCurrentContext();
+        HttpServletRequest request = context.getRequest();
+
+        String mobileNumber = get10DigitMobileNumber(request);
+        
+        log.info("chatbot user is "+ mobileNumber);
+
+        boolean isIsolatedUser = isHomeIsolatedUser(mobileNumber);
+        
+        log.info("isolatedUser "+ isIsolatedUser);
```

Comment:
Why are we putting logging here for production?

Reply:
debug